### DISCUSSION TITLE
using speaker diarization pipeline offline, however program tries to find embedding online

### DISCUSSION THREAD

**Discussion by justinvictor** [pyannote/pyannote-audio#1446] [Q&A] [answered]
I wish to use speaker diarization pipeline offline. however although i managed to point to the offline segmentation model successfully, the program keeps trying to get speechbrain/spkrec-ecapa-voxceleb embedding online.
i downloaded the folder and tried to point the embedding value to the offline folder, however this does not work too.
Any advice is much appreciated, thanks!

pipeline:
  name: pyannote.audio.pipelines.SpeakerDiarization
  params:
    clustering: AgglomerativeClustering
    embedding: speechbrain/spkrec-ecapa-voxceleb
    embedding_batch_size: 32
    embedding_exclude_overlap: true
    segmentation: "C:/VTT/segmentation/pytorch_model.bin"
    segmentation_batch_size: 32

**Comment by justinvictor** [comment, marked as answer]
Found the answer. clone speechbrain/spkrec-ecapa-voxceleb and put it in a folder called speechbrain (the term speechbrain has to appear somewhere in the folder path name). Then the speaker diarization config.yaml file embedding value will now accept a filepath such as  embedding: "C:/VTT/speechbrain/spkrec-ecapa-voxceleb"

**Comment by Kevin-BKQ** [comment]
<img width="835" alt="1739698174923" src="https://github.com/user-attachments/assets/73a9dd62-d6a9-4d09-8ba0-c9c81e2a39cd" />

Is that how you wrote it? But if YAML says this, I will report the following error:

![image](https://github.com/user-attachments/assets/dc5adcc3-c2e8-415b-84a3-3800d0931d41)

Can you help me take a look?
